site/sfguides/src/intro_to_snowpark_container_services/intro_to_snowpark_container_services.md

```diff
@@ -0,0 +1,544 @@
+```bash
+cd .../spcs-101-quickstart/src/jupyter-snowpark
+docker build --platform=linux/amd64 -t <local_repository>/python-jupyter-snowpark:latest .
+```
+Verify the image built successfully:
+```bash
+docker image list
+```
+Now that our local image has built, let's validate that it runs successfully. From a terminal run:
+```bash
+docker run -d -p 8888:8888 <local_repository>/python-jupyter-snowpark:latest
+```
+Open up a browser and navigate to [localhost:8888/lab](http://localhost:8888/lab) to verify your notebook service is working locally. Once you've verified that the service is working, you can stop the container.
+
+### Tag and Push the Image
+Now that we have a local version of our container working, we need to push it to Snowflake so that a Service can access the image. To do this we will create a new tag of the image that points at our image repository in our Snowflake account, and then push said tagged image. From a terminal, run the following:
+```bash
+  # e.g. if repository_url = org-account.registry.snowflakecomputing.com/spcs_hol_db/public/image_repo, snowflake_registry_hostname = org-account.registry.snowflakecomputing.com
+  docker login <snowflake_registry_hostname> -u <user_name>
+  > prompt for password
+  docker tag <local_repository>/python-jupyter-snowpark:latest <repository_url>/python-jupyter-snowpark:dev
+```
+Verify that the new tagged image exists by running:
+```bash
+docker image list
+```
+Now, we need to push our image to Snowflake. From the terminal:
+```bash
+docker push <repository_url>/python-jupyter-snowpark:dev
+```
+This may take some time, so you can move on to the next step **Configure and Push the Spec YAML** while the image is being pushed. Once the `docker push` command completes, you can verify that the image exists in your Snowflake Image Repository by running the following SQL using the Snowflake VSCode Extension or SQL worksheet:
+```sql
+USE ROLE SPCS_USER_ROLE;
+CALL SYSTEM$REGISTRY_LIST_IMAGES('/SPCS_HOL_DB/PUBLIC/IMAGE_REPO');
+```
+You should see your `python-jupyter-snowpark:dev` image listed.
+
+### Configure and Push the Spec YAML
+Services in Snowpark Container Services are defined using YAML files. These YAML files configure all of the various parameters, etc. needed to run the containers within your Snowflake account. These YAMLs support a [large number of configurable parameter](), although we will not reference all of them here. Navigate to your local clone of `.../spcs-101-quickstart/src/jupyter-snowpark/jupyter-snowpark.yaml`, which should look like this:
+```yaml
+spec:
```

Comment:
Do you want to mount the Stage as the Jupyter working directory so that notebooks survive a restart?

Reply:
Yeah I think I'm going to extend it to have a notebook in there, add like a markdown cell change or something, restart the service, and see that the changes are still there.

Reply:
@sfc-gh-bhess I've updated this to both use a volume mount to stage as well as incorporate a Jupyter notebook that leverages the token to run a basic query that calls the other service via the service function